app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -54,9 +54,10 @@ class RecentSyncedTabFeature(
     override fun startLoading() {
         syncStartId?.let { RecentSyncedTabs.recentSyncedTabTimeToLoad.cancel(it) }
         syncStartId = RecentSyncedTabs.recentSyncedTabTimeToLoad.start()
-        store.dispatch(
-            AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.Loading)
-        )
+        // No need to show loading state if a tab is already shown
```

Comment:
nit: Comment is a bit unnecessary - it says the negative of the if-condition but doesn't really say anything we can't already read.